### DISCUSSION TITLE
I have a question about cpm calculations

### DISCUSSION THREAD

**Discussion by Noodate** [apache/skywalking#11458] [General]
For example, the following example talks about the calculation logic of request average time.

![WeChatad95ef32c93a367406865d1643d435c6](https://github.com/apache/skywalking/assets/48311791/12a8e44c-9c80-404a-8377-a1eadb6a407b)

I would like to ask if the throughput rate per minute is also calculated in this way. If it is also calculated according to averaging, then the name of this indicator is called the number of requests per minute. Isn’t it unreasonable?

I have also thought about whether this would be the case. If according to this Period example, at T1, it is directly detected from -T6 to T1 to see whether the number of times that meets the threshold condition is met. If it meets the threshold condition, an alarm will be issued. In this case, it will not It can calculate the average value, which is inconsistent with the calculation logic of the average request time.

So I want to ask, how is cpm calculated?  Many thanks！

**Comment by wu-sheng** [comment]
I think you are mixing things together. Alerting engine is time series for every single minute. Each minute has a value.

CPM is the number of calls in each minute. 

I am not sure which part you are actually asking for.

**Reply by Noodate** [reply]
thanks! Then I understand that cpm is the number of requests per minute, which is different from the calculation of the average time of requests. Is there any document that clearly describes how each indicator is calculated?

**Reply by wu-sheng** [reply]
It is on the UI page of each widget.

**Reply by Noodate** [reply]
got it ，thanks
